packages/plugin-npm/sources/NpmSemverResolver.ts

```diff
@@ -57,6 +59,24 @@ export class NpmSemverResolver implements Resolver {
       try {
         const candidate = new semverUtils.SemVer(version);
         if (range.test(candidate)) {
+          const minimumReleaseAge = opts.project.configuration.get(`npmMinimumReleaseAge`);
+          if (minimumReleaseAge) {
+            const minimumReleaseAgeExclude = opts.project.configuration.get(`npmMinimumReleaseAgeExclude`);
+            const shouldExclude = minimumReleaseAgeExclude.some(exclude =>
+              structUtils.stringifyIdent(descriptor) === exclude
+              || structUtils.stringifyLocator(structUtils.makeLocator(descriptor, version)) === exclude
+              || structUtils.stringifyLocator(structUtils.makeLocator(descriptor, `${PROTOCOL}:${version}`)) === exclude
+              || micromatch.isMatch(structUtils.stringifyDescriptor({...descriptor, range: rawRange}), exclude)
+              || micromatch.isMatch(structUtils.stringifyDescriptor(descriptor), exclude),
```

Comment:
I think I covered the bases here, but let me know if there are other cases to be considered, or any comparison functions that might already exist.

Reply:
I'm not sure. I was thinking users could specify:
1. scope + name - @aws-sdk/types
2. locator, with or without the protocol prefix - @aws-sdk/types@1.2.3 OR @aws-sdk/types@npm:1.2.3
3. descriptor, again, with our without the protocol prefix - @aws-sdk/types@^1.2.3 OR @aws-sdk/types@^npm:1.2.3
4. any glob that would match on the descriptor. so `@aws-sdk/*`, `@aws-sdk/types@*` or even `aws*`.

I think it would be more complicated to condense these down into a narrower set of checks that it would be to just enumerate them.

Reply:
I think I'd prefer to start with those two:

- An ident
- A descriptor with a semver range (and only that; no `npm:` at all), just like peer dependency ranges

I suspect checking on arbitrary descriptors / locators will be more confusing than not, based on the experience we had with the `resolutions` field.